docs(*) add list of annotations used in k8s (#478)
kumahq/kuma-website: docs/docs/1.2.1/documentation/overview.md

Virtual probes annotation enables automatic converting HttpGet probes to virtual. Virtual probe serves on sub-path of insecure port defined in `kuma.io/virtual-probes-port`, i.e `:8080/health/readiness` -> `:9000/8080/health/readiness` where `9000` is a value of `kuma.io/virtual-probes-port` annotation

**Example**

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: example
  annotations:
    kuma.io/virtual-probes: enabled
    kuma.io/virtual-probes-port: "9000"
[...]
```

#### `kuma.io/virtual-probes-port`

Virtual probes port annotation is an insecure port for listening virtual probes

#### `kuma.io/sidecar-env-vars`

Sidecar env vars annotation is a `;` separated list of env vars that will be applied on Kuma Sidecar

**Example**

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: example
  annotations:
    kuma.io/sidecar-env-vars: TEST1=1;TEST2=2 
```

#### `prometheus.metrics.kuma.io/port`

By using this annotation, you can override `Mesh`-wide default port where prometheus should scrape metrics from

**Example**

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: example
  annotations:
    prometheus.metrics.kuma.io/port: "1234"
```

#### `prometheus.metrics.kuma.io/path`

> Metrics prometheus path to override `Mesh`-wide default path where prometheus should scrape metrics from

Reviewer: ```suggestion
This annotation overrides the `Mesh`-wide default path where Prometheus uses to scrape metrics.
```

**Example**

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: example
  annotations:
    prometheus.metrics.kuma.io/path: "/custom-metrics"
```

#### `kuma.io/builtindns`

> Instruct sidecar to use its builtin DNS server

Reviewer: ```suggestion
Instructs the Kuma sidecar to use its builtin DNS server.
```
